src/http/login-http.js

```diff
@@ -0,0 +1,98 @@
+import { createJsonRoute, throwStatus } from "../util/express";
+import crypto from "crypto";
+require("../init-env-variables");
+import * as loginCore from "../core/login-core";
+
+const login = createJsonRoute(function(req, res) {
+  var email = req.body.email;
+  const cipher = crypto.createCipher("aes192", process.env.CRYPTO_PASSWORD);
+  email = cipher.update(email, "utf8", "hex");
+  email += cipher.final("hex");
+  const uuid = req.headers["x-user-uuid"];
+  return loginCore.login(email, uuid);
+});
+
+const addModerator = createJsonRoute(function(req, res, next) {
+  var email = req.body.email;
+  if (!email) {
+    return throwStatus(400, "Email must be provided");
+  }
+  const cipher = crypto.createCipher("aes192", process.env.CRYPTO_PASSWORD);
+  email = cipher.update(email, "utf8", "hex");
+  email += cipher.final("hex");
+  var password = Math.random()
+    .toString(36)
+    .replace(/[^a-z]+/g, "")
+    .substr(0, 30);
+  password = crypto
+    .createHash("md5")
```

Comment:
MD5 is considered cryptographically unsafe. In case our database leaks there is a possibility that user passwords are recovered from the stolen data. It should be pretty straight forward to use SHA-256 instead.

Reply:
Will look into this.

Reply:
Hashing is a good start, but passwords should also be salted with a random salt. This way, users with the same password will still have a different hash stored in the database. I'd use an off the shelf solution like [bcrypt](https://www.npmjs.com/package/bcrypt) or [libsodium](https://www.npmjs.com/package/libsodium). No need to re-invent the wheel here

Reply:
+1 for bcrypt, easy to use and does the job.